File: commonLib/src/main/java/com/travel/commonlib/LoyaltyModel.java
package com.travel.commonlib;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class LoyaltyModel implements Serializable {
    @SerializedName("StatusCode")
    @Expose
    String StatusCode;
    @SerializedName("Message")
    @Expose
    String Message;
    @SerializedName("Data")
    @Expose
    Loyalty_Data data;
    @SerializedName("IsMobileApp")
    @Expose
    boolean isMobileApp;
    @SerializedName("ValidationMessage")
    @Expose
    String validationMessage;
    @SerializedName("AgencyInsufficientBalance")
    @Expose
    boolean agencyInsufficientBalance;
    @SerializedName("vResponse")
    @Expose
    Object v_Response;

    public boolean isMobileApp() {
        return isMobileApp;
    }

    public void setMobileApp(boolean mobileApp) {
        isMobileApp = mobileApp;
    }

    public String getValidationMessage() {
        return validationMessage;
    }

    public void setValidationMessage(String validationMessage) {
        this.validationMessage = validationMessage;
    }

    public boolean isAgencyInsufficientBalance() {
        return agencyInsufficientBalance;
    }

    public void setAgencyInsufficientBalance(boolean agencyInsufficientBalance) {
        this.agencyInsufficientBalance = agencyInsufficientBalance;
    }

    public Object getV_Response() {
        return v_Response;
    }

    public void setV_Response(Object v_Response) {
        this.v_Response = v_Response;
    }

    public String getStatusCode() {
        return StatusCode;
    }

    public void setStatusCode(String statusCode) {
        StatusCode = statusCode;
    }

    public String getMessage() {
        return Message;
    }

    public void setMessage(String message) {
        Message = message;
    }

    public Loyalty_Data getData() {
        return data;
    }

    public void setData(Loyalty_Data data) {
        this.data = data;
    }
}

File: commonLib/src/main/java/com/travel/commonlib/activity/loyality/LoyaltyModel.kt
package com.travel.commonlib.activity.loyality

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serializable

class LoyaltyModel : Serializable {
    @SerializedName("StatusCode")
    @Expose
    var statusCode: String? = null

    @SerializedName("Message")
    @Expose
    var message: String? = null

    @SerializedName("Data")
    @Expose
    var data: Data? = null

    @SerializedName("IsMobileApp")
    @Expose
    var isMobileApp = false

    @SerializedName("ValidationMessage")
    @Expose
    var validationMessage: String? = null

    @SerializedName("AgencyInsufficientBalance")
    @Expose
    var isAgencyInsufficientBalance = false

    @SerializedName("vResponse")
    @Expose
    var v_Response: Any? = null

    class Data {
        @SerializedName("UserId")
        @Expose
        var userId = 0

        @SerializedName("IsApplied")
        @Expose
        var isApplied = false

        @SerializedName("Flag")
        @Expose
        var flag = 0

        @SerializedName("LoyaltyPoints")
        @Expose
        var loyaltyPoints = 0.0

        @SerializedName("LoyaltyValue")
        @Expose
        var loyaltyValue = 0.0

        @SerializedName("RedeemPoints")
        @Expose
        var redeemPoints = 0.0

        @SerializedName("RedeemValue")
        @Expose
        var redeemValue = 0.0

        @SerializedName("Message")
        @Expose
        var message: String? = null
    }
}